nrf_security/src/mbedcrypto_glue/vanilla/CMakeLists.txt

```diff
@@ -2,28 +2,64 @@
 #
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
-if (CONFIG_MBEDTLS_VANILLA_BACKEND)
-  zephyr_library_named(mbedcrypto_glue_vanilla)
-  zephyr_library_sources(
-      ${CMAKE_CURRENT_LIST_DIR}/aes_vanilla.c
-      ${CMAKE_CURRENT_LIST_DIR}/ccm_vanilla.c
-      ${CMAKE_CURRENT_LIST_DIR}/cmac_vanilla.c
-      ${CMAKE_CURRENT_LIST_DIR}/dhm_vanilla.c
-      ${CMAKE_CURRENT_LIST_DIR}/ecdh_vanilla.c
-      ${CMAKE_CURRENT_LIST_DIR}/ecdsa_vanilla.c
-      ${CMAKE_CURRENT_LIST_DIR}/rsa_vanilla.c
-  )
-  
-  zephyr_library_compile_definitions(MBEDTLS_BACKEND_PREFIX=vanilla)
-
-  zephyr_library_include_directories($<TARGET_PROPERTY:${IMAGE}mbedcrypto_glue,INCLUDE_DIRECTORIES>)
-  zephyr_library_compile_definitions($<TARGET_PROPERTY:${IMAGE}mbedcrypto_glue,COMPILE_DEFINITIONS>)
-  add_custom_command(
-    TARGET ${IMAGE}mbedcrypto_glue_vanilla
-    POST_BUILD
-    COMMAND ${CMAKE_OBJCOPY} 
-            --redefine-syms
-            ${CMAKE_CURRENT_BINARY_DIR}/symbol_rename_vanilla.txt
-            $<TARGET_FILE:${IMAGE}mbedcrypto_glue_vanilla>
-  )
-endif()
+
+#
+# Create vanilla noglue library
+# Doesn't have ALT flags set (for correct context size)
+# This implements features that are to be glued (but with context-sizes that are corresponding)
+# This must use be compiled with its own config-file
+# This library must not be linked with directly
+#
+append_with_prefix_ifdef(CONFIG_VANILLA_MBEDTLS_AES_C   src_vanilla
+  ${ARM_MBEDTLS_ROOT}/library/ aes.c
+)
+append_with_prefix_ifdef(CONFIG_VANILLA_MBEDTLS_CCM_C   src_vanilla
+  ${ARM_MBEDTLS_ROOT}/library/ ccm.c
+)
+append_with_prefix_ifdef(CONFIG_VANILLA_MBEDTLS_RSA_C   src_vanilla
+  ${ARM_MBEDTLS_ROOT}/library/ rsa.c
+)
```

Comment:
I don't get it. Why construct it with a prefix? Why not just:

list_append_ifdef(CONFIG_VANILLA_MBEDTLS_RSA_C src_vanilla ${ARM_MBEDTLS_ROOT}/library/rsa.c)

Reply:
Or even better,

zephyr_library_sources_ifdef(CONFIG_VANILLA_MBEDTLS_RSA_C ${ARM_MBEDTLS_ROOT}/library/rsa.c)

Reply:
Fixed  (changed to list_append_ifdef)

Reply:
zephyr_library_sources_ifdef is better.

Reply:
Won't fix. I'm printing the files in the list when debug flag is set

Reply:
Why do we need to do that?

We want the build scripts to be consistent so that they are easier to read.

Reply:
Fixed after some discussions (found a better way to show debug)